src/harness/harnessIO.ts

```diff
@@ -87,19 +87,18 @@ function createNodeIO(): IO {
     function listFiles(path: string, spec: RegExp, options: { recursive?: boolean; } = {}) {
         function filesInFolder(folder: string): string[] {
             let paths: string[] = [];
+            for (const entry of fs.readdirSync(folder, { withFileTypes: true })) {

```

Comment:
Apply the same optimization to `getAccessibleFileSystemEntries`?

Reply:
Yeah, this is a good idea; the implementation we have in `sys.ts` already uses `withFileTypes`.

(That being said, this whole block just duplicates functionality in `sys.ts` so I do wonder how much copy pasting we could avoid.)